recipe-server/normandy/recipes/models.py

```diff
@@ -373,6 +393,52 @@ def save(self, *args, **kwargs):
         self.implementation_hash = self.compute_implementation_hash()
         super().save(update_fields=['implementation_hash'])
 
+    def validate_arguments(self, arguments, old_arguments=None):
+        """
+        Test if `arguments` follows all action-specific rules.
+
+        If there is a previous version of the arguments, they should
+        be passed as `old_arguments`. Some validation rules depend on
+        the history of the arguments.
```

Comment:
Technically, they don't. They only rely on knowing whether the recipe arguments being validated are for a new or existing recipe. Although I know the intent is to eventually use `old_arguments` in that way.

Earlier we talked about this being a method on the `Recipe` class and using the dirty mixin for getting old values for comparison, which would give you access to the info from `old_arguments` without having to pass it in. Why'd you move this to the action class instead?